src/interface/data_collection.cpp

```diff
@@ -70,15 +70,13 @@ GetOrAdd_impl(Mesh *pmy_mesh_,
     if (partitions.size() == 0) partitions = std::vector<BlockList_t>(1);
     for (auto i = 0; i < partitions.size(); i++) {
       std::string md_label = mbd_label + "_part-" + std::to_string(i);
-      if (gmg_level >= 0) md_label = md_label + "_gmg-" + std::to_string(gmg_level);
+      if (gmg_level) md_label = md_label + "_gmg-" + std::to_string(*gmg_level);
       containers_[md_label] = std::make_shared<MeshData<Real>>(mbd_label);
       containers_[md_label]->Set(partitions[i], pmy_mesh_);
-      if (gmg_level >= 0) {
-        int min_gmg_logical_level = pmy_mesh_->GetGMGMinLogicalLevel();
-        containers_[md_label]->grid = GridIdentifier{GridType::two_level_composite,
-                                                     gmg_level + min_gmg_logical_level};
+      if (gmg_level) {
+        containers_[md_label]->grid = GridIdentifier::two_level_composite(*gmg_level);
       } else {
-        containers_[md_label]->grid = GridIdentifier{GridType::leaf, 0};
+        containers_[md_label]->grid = GridIdentifier::leaf();
```

Comment:
gmg levels can now be negative, so I switch to a `std::optional` for `gmg_level`. If it is not set, it implies that we are working with meshdata for the regular leaf grid.